helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -646,10 +645,9 @@ private void subscribeForChanges(NotificationContext.Type callbackType, String p
               break;
             }
             default: {
-              List<String> childNames = _zkClient.getChildren(path);
-              if (childNames != null) {
-                for (String childName : childNames) {
-                  String childPath = path + "/" + childName;
+              if (children != null) {
```

Comment:
There is a slightly behavior difference here. Originally before this change, if type is Finalize, we still iterate trough all node and do `subscribeDataChange `. 
Now we do not return a children list when finalize (line 565) so we do not do `subscribeDataChange` for those nodes.  
This behavior difference cause unit test `TestListenerCallbackBatchMode` fail on my side. Did you run in to this problem?
Thx.

Reply:
Yeah, I was aware of the failed test and the code in `subscribeChildChange()` for `NotificationContext.Type.FINALIZE`.